processor/src/main/java/org/mapstruct/ap/internal/model/MappingMethod.java

```diff
@@ -238,9 +238,6 @@ public boolean equals(Object obj) {
             return false;
         }
         final MappingMethod other = (MappingMethod) obj;
-        if ( (this.name == null) ? (other.name != null) : !this.name.equals( other.name ) ) {
```

Comment:
I think that this was a bug that we had before. Whenever we forge methods we can reuse mappings if they are the same. However, if we take the name into consideration, they'll never be the same, because we create save method names.